mlrun/api/db/sqldb/db.py

```diff
@@ -3916,3 +3918,104 @@ def set_run_notifications(
                 project=project,
             )
         self._commit(session, [run], ignore=True)
+
+    def store_datastore_profile(
+        self, session, info: mlrun.common.schemas.DatastoreProfile
+    ):
+        """
+        Create or replace a datastore profile.
+        :param session: SQLAlchemy session
+        :param info: datastore profile
+        :returns: None
+        """
+        info.project = info.project or config.default_project
+        profile = self._query(
+            session, DatastoreProfile, name=info.name, project=info.project
+        )
+        first = profile.first()
+        if first:
+            first.type = info.type
+            first.body = info.body
+            self._commit(session, [profile])
```

Comment:
Should be `[first]`.
Anyway, you should call `_upsert` here as well.

Reply:
_upsert) calls session.add(), which should not be called on update I believe